cpp/src/arrow/ipc/json-test.cc

```diff
@@ -57,6 +57,11 @@ void TestSchemaRoundTrip(const Schema& schema) {
   writer.EndObject();
 
   std::string json_schema = sb.GetString();
+  // RapidJSON can use SIMD operations that read past string
```

Comment:
Are you sure about it? Is there a reference somewhere in the RapidJSON documentation and/or did you ask them?

Reply:
Ok... It seems there are [two versions](https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/reader.h#L280) of whitespace-skipping internally.
* one with a `(const char* p, const char* end)` signature that is careful not to read past the `end` pointer
* one with a `(const char* p)` signature which seems to be reading past the terminating NUL character

Unfortunately, which one is called seems to depend which high-level API is used exactly (see https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/reader.h#L502-L514).

Reply:
uses parse(char*, length)